src/AllLayers.js

```diff
@@ -71,38 +71,81 @@ L.LayerGroup.environmentalLayers = L.LayerGroup.extend(
     onAdd: function(map) {
       this._map = map;
       this.overlayMaps = {};
+      this.groupedOverlayMaps = {}; // For grouping layers in the new menu
       var baseMaps = this.options.baseLayers ? this.options.baseLayers : { "Grey-scale": this.options.defaultBaseLayer.addTo(map) };
 
       for (let layer of this.options.layers.include) {
         if (this.options.layers0.includes(layer)) {
           this.overlayMaps[layer] = window['L']['layerGroup'][layer]();
+          if (layer === 'purpleLayer' && !this.groupedOverlayMaps.PurpleAir) {
```

Comment:
Hey this is getting messy, can we use switch case instead?

Reply:
Sure, will do.